tiledb/arrow/src/datatype.rs

```diff
@@ -130,10 +130,60 @@ pub fn is_same_physical_type(
 }
 
 #[cfg(test)]
-mod tests {
+pub mod tests {
     use super::*;
     use proptest::prelude::*;
 
+    /// Returns a strategy for generating any Arrow data type
+    /// which corresponds to a TileDB Datatype
+    pub fn arbitrary_arrow_invertible(
+    ) -> impl Strategy<Value = arrow_schema::DataType> {
+        use arrow_schema::DataType as DT;
+        use arrow_schema::TimeUnit as TU;
+        prop_oneof![
+            Just(DT::Int8),
+            Just(DT::Int16),
+            Just(DT::Int32),
+            Just(DT::Int64),
+            Just(DT::UInt8),
+            Just(DT::UInt16),
+            Just(DT::UInt32),
+            Just(DT::UInt64),
+            Just(DT::Float32),
+            Just(DT::Float64),
+            Just(DT::Timestamp(TU::Second, None)),
+            Just(DT::Timestamp(TU::Millisecond, None)),
+            Just(DT::Timestamp(TU::Microsecond, None)),
+            Just(DT::Timestamp(TU::Nanosecond, None)),
+            Just(DT::Time64(TU::Microsecond)),
+            Just(DT::Time64(TU::Nanosecond))
+        ]
+    }
+
+    pub fn arbitrary_arrow_implemented(
+    ) -> impl Strategy<Value = arrow_schema::DataType> {
+        tiledb_test::datatype::arbitrary_implemented().prop_map(|dt| arrow_type_physical(&dt).expect("Datatype claims to be implemented but does not have an arrow equivalent"))
```

Comment:
How in the world is `cargo fmt` ok with this line?

Reply:
It's a pretty excellent question. I have no idea.  The formatter doesn't touch it if I manually tweak it so I'll just do that.